aws-http-nio-client-netty/src/it/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClientIntegrationTest.java

```diff
@@ -84,6 +89,60 @@ public static void tearDown() throws Exception {
     }
 
     @Test
+    public void customFactoryIsUsed() throws Exception {
+        ThreadFactory threadFactory = spy(new CustomThreadFactory());
+        SdkAsyncHttpClient customClient =
+                NettySdkHttpClientFactory.builder()
+                                         .trustAllCertificates(true)
+                                         .eventLoopGroup(EventLoopGroupBuilder.builder()
+                                                                              .threadFactory(threadFactory)
+                                                                              .build())
+                                         .build()
+                                         .createHttpClient();
+
+        makeSimpleRequest(customClient);
+
+        Mockito.verify(threadFactory, atLeastOnce()).newThread(Mockito.any());
+    }
+
+    @Test
+    public void customThreadCountIsRespected() throws Exception {
+        final int threadCount = 10;
+        ThreadFactory threadFactory = spy(new CustomThreadFactory());
+        SdkAsyncHttpClient customClient =
+                NettySdkHttpClientFactory.builder()
+                                         .trustAllCertificates(true)
+                                         .eventLoopGroup(EventLoopGroupBuilder.builder()
+                                                                              .numberOfThreads(threadCount)
+                                                                              .threadFactory(threadFactory)
+                                                                              .build())
+                                         .build()
+                                         .createHttpClient();
+
+        // Have to make enough requests to prime the threads
+        for (int i = 0; i < threadCount; i++) {
```

Comment:
Does this test that the thread count is respected? Wouldn't we want to make many more requests than the thread count an ensure that no more than 10 (or whatever thread count is set as) threads are created?